components/service/experiments/src/main/java/mozilla/components/service/experiments/Experiments.kt

```diff
@@ -314,6 +364,7 @@ open class ExperimentsInternalAPI internal constructor() {
     }
 }
 
+@SuppressLint("StaticFieldLeak")
```

Comment:
As we can't not enforce by the compiler, that the `context` pass to `initialize` function is a `applicationContext` I think could be a good idea at least, warm the consumers in the docs, about possible memory leaks if the don't comply.

``` kotlin
    /**
     * Initialize the experiments library.
     *
     * This should only be initialized once by the application.
     *
     * @param applicationContext [Context] to access application features, such
     * as shared preferences. No passing an applicationContext here could cause memory leaks.
     */
    fun initialize(
        applicationContext: Context,
        configuration: Configuration = Configuration()
    ){}
```

Reply:
Expanded the comment to reflect this, thanks!